frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts

```diff
@@ -70,11 +84,62 @@ export class FeatureFlagOverviewDetailsSectionCardComponent {
     this.dialogService.openDisableFeatureFlagConfirmModel();
   }
 
-  onMenuButtonItemClick(event) {
-    if (event === 'Delete') {
-      this.dialogService.openDeleteFeatureFlagModal();
-    } else if (event === 'Edit') {
-      this.dialogService.openEditFeatureFlagModal();
+  onMenuButtonItemClick(event: FEATURE_FLAG_DETAILS_PAGE_ACTIONS, flag: FeatureFlag) {
+    switch (event) {
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DELETE:
+        this.dialogService.openDeleteFeatureFlagModal();
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EDIT:
+        this.dialogService.openEditFeatureFlagModal();
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.DUPLICATE:
+        console.log('Duplicate feature flag');
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.ARCHIVE:
+        console.log('Archive feature flag');
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EXPORT_DESIGN:
+        this.openConfirmExportDesignModal();
+        break;
+      case FEATURE_FLAG_DETAILS_PAGE_ACTIONS.EMAIL_DATA:
+        this.openConfirmEmailDataModal();
+        break;
+      default:
+        console.log('Unknown action');
+    }
+  }
+
+  openConfirmExportDesignModal() {
+    const confirmMessage = 'feature-flags.export-feature-flag-design.confirmation-text.text';
+    this.dialogService.openExportFeatureFlagDesignModal(confirmMessage)
+      .afterClosed()
+      .subscribe((isEmailClicked: boolean) => {
+        if (isEmailClicked) {
+          this.featureFlagService.exportFeatureFlagsData([this.featureFlag.id]);
+        }
+      });
+  }
+
+  openConfirmEmailDataModal() {
+    const confirmMessage = 'feature-flags.export-feature-flags-data.confirmation-text.text';
+    const emailConfirmationMessage = "The feature flag will be sent to '" + this.emailId + "'." ;
+    this.dialogService.openEmailFeatureFlagDataModal(confirmMessage, emailConfirmationMessage)
+      .afterClosed()
+      .subscribe((isEmailClicked: boolean) => {
+        if (isEmailClicked) {
+          this.featureFlagService.emailFeatureFlagData(this.featureFlag.id);
+        }
+      });
+  }
+
+  async getUserEmail() {
```

Comment:
we can simplify without any Promises or subscriptions. selectors and async template pipe should allow us to pass the email address directly into the `openConfirmEmailDataModal()` function.

In `auth.selectors` create a selector for the email address of current user:

```
export const selectUserEmail = createSelector(
  selectAuthState,
  (state) => state.user?.email || ''
);
```

Then use that selector in featureFlag service and reference in component:

`feature-flag.service`
```
this.currentUserEmailAddress$ = this.store.pipe(select(selectUserEmail));
```

`feature-flag.component.ts`
```
this.emailId$ = this.featureFlagService.currentUserEmailAddress$
```

`feature-flag.component.html`
```
<button (click)="openConfirmEmailDataModal((emailId$ | async))">Send Email</button>
```

and then finally back in the `component.ts`:

```
openConfirmEmailDataModal(emailId: string) {
    const confirmMessage = 'feature-flags.export-feature-flags-data.confirmation-text.text';
    const emailConfirmationMessage = `The feature flag will be sent to '${emailId}'.`;
    this.dialogService.openEmailFeatureFlagDataModal(confirmMessage, emailConfirmationMessage)
      .afterClosed()
      .subscribe((isEmailClicked: boolean) => {
        if (isEmailClicked) {
          this.featureFlagService.emailFeatureFlagData(this.featureFlag.id);
        }
      });
  }
```

Reply:
BTW I have not tried this code out, it may need some adjustments but I hope the pattern is clear. one thing I'm learning about ngrx as we work on this is that `selectors` + `async` pipes in the template solve a lot of problems concerning getting data "out" of the data store.

Reply:
The problem with this is that we can't pass email from HTML, we aren't calling open emailDataModal directly. We have to extract value from observable through subscriber or `firstvalue` (this will need async func).